tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -0,0 +1,756 @@
+		}
+	}
+	nicID := vmNic.ID
+
+	networkProfile = &compute.NetworkProfile{
+		NetworkInterfaces: &[]compute.NetworkInterfaceReference{
+			{
+				ID: nicID,
+				NetworkInterfaceReferenceProperties: &compute.NetworkInterfaceReferenceProperties{
+					Primary: to.BoolPtr(true),
+				},
+			},
+		},
+	}
+	return networkProfile, nil
+}
+
+// CreateWindowsVM takes in imageId, instanceType and sshKey name to create Windows instance under the same
+// resourceGroupName as the existing OpenShift
+// TODO: If it fails during the instance creation process it has to delete the resources created
+// untill that step.
+func (az *AzureProvider) CreateWindowsVM(imageId, instanceType, sshKey string) (err error) {
+	// Construct the VirtualMachine properties
+	rand.Seed(time.Now().UnixNano())
+	ctx := context.Background()
+	vmHardwareProfile := &compute.HardwareProfile{
+		VMSize: compute.VirtualMachineSizeTypes(instanceType)}
+	log.Info(fmt.Sprintf("constructed the HardwareProfile for node"))
+
+	vmStorageProfile := az.constructStorageProfile(imageId)
+	log.Info(fmt.Sprintf("constructed the Storage Profile for node"))
+
+	vmOSProfile, instanceName, adminPassword := az.constructOSProfile(ctx)
+	log.Info(fmt.Sprintf("constructed the OSProfile for the node"))
+
+	vmNetworkProfile, err := az.constructNetworkProfile(ctx, instanceName)
+	log.Info(fmt.Sprintf("constructed the network profile for the node"))
+
+	log.Info(fmt.Sprintf("constructed all the profiles, about to create instance."))
+	nodeLocation := *(az.getvnetProfile(ctx).Location)
+	future, err := az.vmClient.CreateOrUpdate(
+		ctx,
+		az.resourceGroupName,
+		instanceName,
+		compute.VirtualMachine{
+			Location: to.StringPtr(nodeLocation),
+			VirtualMachineProperties: &compute.VirtualMachineProperties{
+				HardwareProfile: vmHardwareProfile,
+				StorageProfile:  vmStorageProfile,
+				OsProfile:       vmOSProfile,
+				NetworkProfile:  vmNetworkProfile,
+			},
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("instance failed to create: %v", err)
+	}
+	err = future.WaitForCompletionRef(ctx, az.vmClient.Client)
+	if err != nil {
+		return fmt.Errorf("instance failed to create: %v", err)
+	}
+	vmInfo, err := future.Result(az.vmClient)
+	check(err, "failed to obtain instance info")
+	resourceTrackerFilePath, err := resource.MakeFilePath(az.resourceTrackerDir)
+	check(err, "unable to create file")
+	if az.NsgName == "" {
+		resource.AppendInstallerInfo([]string{*(vmInfo.Name)}, []string{}, resourceTrackerFilePath)
+	} else {
+		resource.AppendInstallerInfo([]string{*(vmInfo.Name)}, []string{az.NsgName}, resourceTrackerFilePath)
+	}
+
+	log.Info(fmt.Sprintf("Successfully created windows instance: %s", instanceName))
+	ipAddress, ipErr := az.getIPAddress(ctx)
+	check(ipErr, fmt.Sprintf("couldn't get the IP Address for corresponding resource: %s", az.IpName))
+	if ipErr != nil {
+		ipAddress = az.IpName
+	}
+	resultPath := az.resourceTrackerDir + "/" + instanceName
```

Comment:
Shouldn't the file name be `windows-installer.json` and not the instance you created. The file should contain information related to resources that were created, if you want to include the username and password to access that file then shouldn't you generate a new file?

Reply:
Yep taken care of that.